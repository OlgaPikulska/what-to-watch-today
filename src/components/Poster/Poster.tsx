import React from "react";
import Image from "next/image";
import NoPhotographyOutlinedIcon from "@mui/icons-material/NoPhotographyOutlined";
import { Movie } from "@/types";

type Poster = Pick<Movie, "poster_path" | "title">;

export const Poster = ({ poster_path, title }: Poster) => {
	return (
		<>
			{poster_path ? (
				<Image
					src={`https://image.tmdb.org/t/p/w200${poster_path}`}
					alt={`Poster of ${title} movie`}
					className="h-[300px] w-[200px] rounded-sm"
					width={200}
					height={300}
				/>
			) : (
				<div className="flex h-[300px] w-[200px] items-center justify-center rounded-sm border border-gray-500 text-gray-500">
					<NoPhotographyOutlinedIcon fontSize="large" />
					<span className="mt-2 text-sm">No Poster</span>
				</div>
			)}
		</>
	);
};
