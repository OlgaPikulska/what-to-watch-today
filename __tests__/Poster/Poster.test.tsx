import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import { Poster } from "@/components/Poster/Poster";

const mockPosterPath = "mockPoster.jpg";
const mockTitle = "Mock Movie";

describe("Poster Component", () => {
	it("should matches the snapshot when poster_path is provided", () => {
		const { container } = render(<Poster poster_path={mockPosterPath} title={mockTitle} />);
		expect(container).toMatchSnapshot();
	});

	it("should matches the snapshot when poster_path is not provided", () => {
		const { container } = render(<Poster poster_path="" title={mockTitle} />);
		expect(container).toMatchSnapshot();
	});

	it("should render the Image component when poster_path is provided", () => {
		const { getByAltText } = render(<Poster poster_path={mockPosterPath} title={mockTitle} />);
		const imageElement = getByAltText(`Poster of ${mockTitle} movie`);
		expect(imageElement).toBeInTheDocument();
		expect(imageElement).toHaveAttribute(
			"src",
			`/_next/image?url=https%3A%2F%2Fimage.tmdb.org%2Ft%2Fp%2Fw200${mockPosterPath}&w=640&q=75`,
		);
		expect(imageElement).toHaveClass("h-[300px] w-[200px] rounded-sm");
	});

	it("should render fallback UI when poster_path is not provided", () => {
		const { getByTestId, getByText } = render(<Poster poster_path="" title={mockTitle} />);

		const iconElement = getByTestId("NoPhotographyOutlinedIcon");
		expect(iconElement).toBeInTheDocument();

		const textElement = getByText("No Poster");
		expect(textElement).toBeInTheDocument();
	});
});
