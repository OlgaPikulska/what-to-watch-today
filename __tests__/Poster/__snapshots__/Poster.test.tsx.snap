// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Poster Component should matches the snapshot when poster_path is not provided 1`] = `
<div>
  <div
    class="flex h-[300px] w-[200px] items-center justify-center rounded-sm border border-gray-500 text-gray-500"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge css-uks7c6-MuiSvgIcon-root"
      data-testid="NoPhotographyOutlinedIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M8.9 6.07 7.48 4.66 9 3h6l1.83 2H20c1.1 0 2 .9 2 2v12c0 .05-.01.1-.02.16L20 17.17V7h-4.05l-1.83-2H9.88zm11.59 17.24L18.17 21H4c-1.1 0-2-.9-2-2V7c0-.59.27-1.12.68-1.49l-2-2L2.1 2.1l19.8 19.8zM9.19 12.02c-.11.31-.19.63-.19.98 0 1.66 1.34 3 3 3 .35 0 .67-.08.98-.19zM16.17 19l-1.68-1.68c-.73.43-1.58.68-2.49.68-2.76 0-5-2.24-5-5 0-.91.25-1.76.68-2.49L4.17 7H4v12zm-1.36-7.02 2.07 2.07Q17 13.54 17 13c0-2.76-2.24-5-5-5-.36 0-.71.04-1.06.12l2.07 2.07c.84.3 1.5.96 1.8 1.79"
      />
    </svg>
    <p
      class="MuiTypography-root MuiTypography-body2 mt-2 css-7ctihp-MuiTypography-root"
    >
      No Poster
    </p>
  </div>
</div>
`;

exports[`Poster Component should matches the snapshot when poster_path is provided 1`] = `
<div>
  <img
    alt="Poster of Mock Movie movie"
    class="h-[300px] w-[200px] rounded-sm"
    data-nimg="1"
    decoding="async"
    height="300"
    loading="lazy"
    src="/_next/image?url=https%3A%2F%2Fimage.tmdb.org%2Ft%2Fp%2Fw200mockPoster.jpg&w=640&q=75"
    srcset="/_next/image?url=https%3A%2F%2Fimage.tmdb.org%2Ft%2Fp%2Fw200mockPoster.jpg&w=256&q=75 1x, /_next/image?url=https%3A%2F%2Fimage.tmdb.org%2Ft%2Fp%2Fw200mockPoster.jpg&w=640&q=75 2x"
    style="color: transparent;"
    width="200"
  />
</div>
`;
